sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -338,10 +348,12 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
           case Some(_: CudaFatalException) =>
             logError("Stopping the Executor based on exception being a fatal CUDA error: " +
               s"${ef.toErrorString}")
+            logGpuDebugInfo()
             System.exit(20)
           case Some(_: CudaException) =>
-            logDebug(s"Executor onTaskFailed because of a non-fatal CUDA error: " +
+            logWarning(s"Executor onTaskFailed because of a non-fatal CUDA error: " +
               s"${ef.toErrorString}")
+            logGpuDebugInfo()
```

Comment:
since we could have many tasks failing at the same time (or queued up waiting to fail due to this..) maybe we should set a static variable that says this info was printed once? I am actually pretty confused why this isn't a fatal error. Or maybe this gets printed once a minute or something (if tasks keep failing sporadically every so many minutes?)

Reply:
Good point.  This also goes to a [question I asked in the issue](https://github.com/NVIDIA/spark-rapids/issues/5028#issuecomment-1255597384).  Do we want to do this only for cudaFatalExceptions, in which case we are exiting and need to just print it.   If we want this output for non-fatal cuda exceptions as well, then I agree we need to put in some kind of limiting.  Maybe just once would be ok in that case?  Or once every x task failures.

Reply:
I assume/hope "ai.rapids.cudf.CudaException: unknown error" is a fatal exception (from the original issue). If it is then yeah I would focus on this for fatal exception only.

For other CUDA exceptions, I am not sure the nvidia-smi output helps. It seems you really want to know what is going on in the host if we can't attach to it like in the "unknown error" case.

Reply:
Another question - I changed the logDebug here to a logWarning - seems like we'd want to see these.  Was there a reason it was logDebug for the non-fatal failures?  Log output reduction?